### PR TITLE
add package test case

### PR DESCRIPTION
**Pull request by junxiawang**
@immarvin , @tingtli , @hu-weihua  please help review os package and other package test case

### CODE REVIEW

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 1]
```diff
@@ -1,69 +1,111 @@
...
+cmd:perl /opt/xcat/share/xcat/tools/autotest/testcase/pkg/pkgtest.pl -s $$osimage pkglist __GETNODEATTR($$CN,os)__
 check:rc==0
-cmd:rm -rf /opt/xcat/share/xcat/install/rh/compute.*.pkglist.1
-cmd:xdsh $$CN rpm -qa |grep ntp-4.2
+cmd:updatenode $$CN -S
+check:rc==0
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN dpkg -l |grep "ii  rpm"|grep "package manager for RPM" ;else xdsh $$CN rpm -qa |grep xCAT-test; fi
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: why we use different output check for ubuntu and rh/sles?
why didn't you check xcat-test was installed on ubuntu?

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 2]
```diff
@@ -1,69 +1,111 @@
...
+cmd:updatenode $$CN -S
+check:rc==0
+cmd:perl /opt/xcat/share/xcat/tools/autotest/testcase/pkg/pkgtest.pl -c $$osimage pkglist __GETNODEATTR($$CN,os)__
+cmd:rc==0
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN dpkg -l |grep "ii  rpm" |grep "package manager for RPM";else xdsh $$CN rpm -qa |grep xCAT-test; fi
 check:rc!=0
-cmd:if cat /etc/*-release | grep SUSE >/dev/null; then ver=`echo "$$os"  |awk -F "." '{print $1}'`; `yes| cp -f /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist.bak /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist`;elif  cat /etc/*release |grep "Red Hat" >/dev/null;then ver=`echo "$$os"  |awk -F "." '{print $1}'`;`yes |cp -f /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist.bak  /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist`;fi
-#cmd:rm -rf /opt/xcat/share/xcat/install/rh/compute.*.pkglist.1
-cmd:rm -rf /opt/xcat/share/xcat/install/rh/compute.*.pkglist.bak
+cmd:copycds $$ISO
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: what is the goal to copy the iso here?

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 3]
```diff
@@ -1,69 +1,111 @@
...
-cmd:if cat /etc/*-release | grep SUSE >/dev/null; then ver=`echo "$$os"  |awk -F "." '{print $1}'`; cp /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist.bak;  `cat /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist |sed 's/ntp/-ntp/g' >> /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist.1`;`yes | cp -f   /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist.1  /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist` ;updatenode $$CN -S;elif  cat /etc/*release |grep "Red Hat" >/dev/null;then ver=`echo "$$os"  |awk -F "." '{print $1}'`;cp /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist.bak;`cat /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist |sed 's/ntp/-ntp/g'  >>/opt/xcat/share/xcat/install/rh/compute.$ver.pkglist.1`;`yes |cp -f /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist.1 /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist` ;updatenode $$CN -S;fi
-cmd:rm -rf /opt/xcat/share/xcat/install/rh/compute.*.pkglist.1
+cmd:perl /opt/xcat/share/xcat/tools/autotest/testcase/pkg/pkgtest.pl -s $$osimage pkglist  __GETNODEATTR($$CN,os)__
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: I think here you need to copycds first, right?

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 4]
```diff
@@ -1,69 +1,111 @@
...
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN  dpkg -r rpm; else  xdsh $$CN rpm -e xCAT-test; fi
+check:rc==0
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN dpkg -l |grep "ii  rpm"|grep "package manager for RPM" ;else xdsh $$CN rpm -qa |grep xCAT-test; fi
 check:rc!=0
-cmd:if cat /etc/*-release | grep SUSE >/dev/null; then ver=`echo "$$os"  |awk -F "." '{print $1}'`; `yes| cp -f /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist.bak /opt/xcat/share/xcat/install/sles/compute.$ver.pkglist`;elif  cat /etc/*release |grep "Red Hat" >/dev/null;then ver=`echo "$$os"  |awk -F "." '{print $1}'`;`yes |cp -f /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist.bak  /opt/xcat/share/xcat/install/rh/compute.$ver.pkglist`;fi
-#cmd:rm -rf /opt/xcat/share/xcat/install/rh/compute.*.pkglist.1
+cmd:copycds $$ISO
+check:rc==0
+end
+start:add_includeospkg_updatenode
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: should the name be add_include_ospkg_updatenode?

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 5]
```diff
@@ -1,69 +1,111 @@
...
+descreption:add include ospkg list
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: is here add #INCLUDE?

File: xCAT-test/autotest/testcase/pkg/cases0 [hunk 6]
```diff
@@ -1,69 +1,111 @@
...
+cmd:perl /opt/xcat/share/xcat/tools/autotest/testcase/pkg/pkgtest.pl -c  $$osimage pkglist  __GETNODEATTR($$CN,os)__
+check:rc==0
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN  dpkg -r rpm; else  xdsh $$CN rpm -e xCAT-test; fi
+check:rc==0
+cmd:if echo "__GETNODEATTR($$CN,os)__" | grep ubuntu >/dev/null; then xdsh $$CN dpkg -l |grep "ii  rpm"|grep "package manager for RPM" ;else xdsh $$CN rpm -qa |grep xCAT-test; fi
+check:rc!=0
+cmd:copycds $$ISO
+check:rc==0
 end
 start:add_other_pkg
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/pkg/cases0]: add_otherpkg_updatenode?

File: xCAT-test/autotest/testcase/pkg/pkgtest.pl [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+# IBM(c) 2007 EPL license http://www.eclipse.org/legal/epl-v10.html
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Data::Dumper; 
+use Term::ANSIColor;
+use Time::Local;
+BEGIN
+{
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/pkg/pkgtest.pl]: The BEGIN should better be put above ``use`` lines

File: xCAT-test/autotest/testcase/pkg/pkgtest.pl [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    print "\n";
+    return;
+}
+
+sub getimgattr
+{
+    my @output = `lsdef -t osimage -o  $osimg -i $attr`;
+    my $pkglistvalue;
+    print " output is @output \n";   
+    if($?){
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/pkg/pkgtest.pl]: what's this for?

Comment (junxiawang) [inline xCAT-test/autotest/testcase/pkg/pkgtest.pl]: To get osimg's attributes .
Such as pkglist otherpkgdir's value

**Comment by tingtli** [comment]
@junxiawang Please update the test cases according to yangsong's and my comments. Thanks!

**Comment by junxiawang** [comment]
@tingtli  done accongding to your and yangsong's comments

**Comment by zet809** [comment]
Hi, @junxiawang , any updates for this document?

**Comment by whowutwut** [comment]
This PR has been open for so long, are we sure it's still needed?

**Comment by tingtli** [comment]
i have reviewed and agree to merge.

**Pull request merged**